src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -1364,6 +1425,110 @@ MultiPlanTree(Query *queryTree)
 
 
 /*
+ * HasUnsupportedReferenceTableJoin returns true if there exists a outer join
+ * between reference table and distributed tables which does not follow
+ * the rules :
+ * - Reference tables can not be located in the outer part of the semi join (or
+ * the inner part of the anti join). Otherwise, we may have duplicate results.
+ * Although getting duplicate results is not possible by checking the equality
+ * on the column of the reference table and partition column of distributed table,
+ * we still keep these checks. Because, using the reference table in the outer
+ * part of the semi join is not very common.
+ * - Reference tables can not be located in the outer part of the left join.
+ * Otherwise we will definitely have duplicate rows. Beside, reference tables
+ * can not be used with full outer joins because of the same reason.
+ */
+static bool
+HasUnsupportedReferenceTableJoin(PlannerRestrictionContext *plannerRestrictionContext)
+{
+	List *joinRestrictionList =
+		plannerRestrictionContext->joinRestrictionContext->joinRestrictionList;
+	ListCell *joinRestrictionCell = NULL;
+
+	foreach(joinRestrictionCell, joinRestrictionList)
+	{
+		JoinRestriction *joinRestriction = (JoinRestriction *) lfirst(
+			joinRestrictionCell);
+		JoinType joinType = joinRestriction->joinType;
+		PlannerInfo *plannerInfo = joinRestriction->plannerInfo;
+		RelOptInfo *innerrel = joinRestriction->innerrel;
+		RelOptInfo *outerrel = joinRestriction->outerrel;
+
+		if (joinType == JOIN_SEMI || joinType == JOIN_ANTI || joinType == JOIN_LEFT)
```

Comment:
I know JOIN_RIGHT is never received here. Might make sense to add it for completeness purposes